### PR TITLE
Enable convolution_backward with bias and symints

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85970

Originally by Krovatkin from https://github.com/pytorch/pytorch/pull/85816

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: tools/autograd/load_derivatives.py [hunk 1]
```diff
@@ -781,6 +781,19 @@ def stride_expr(name: str) -> str:
                 ),
                 "expr": lambda name: f"{name}.has_value() ? c10::optional<IntArrayRef>({name}->sizes()) : c10::nullopt",
             },
+
+        ),
+        # replace self->sym_sizes() with self_sym_sizes_opt
+        (
+            r"{}->sym_sizes\(\)",
```
Comment (albanD) [inline tools/autograd/load_derivatives.py:788]: Wait, we already replace sym_sizes() into self_sym_sizes two rules above :p 
One of the two needs to be removed.

File: torchgen/api/translate.py [hunk 2]
```diff
@@ -350,7 +351,16 @@ def direct_solve(goal: NamedCType) -> str:
             )
             return f"{argname}.has_value() ? c10::make_optional({argname}->expect_int()) : c10::nullopt"
         elif goal.type == BaseCType(optionalIntArrayRefT):
-            return direct_solve(NamedCType(goal.name, optionalLongVec_ctype))
+            try:
+                argname = direct_solve(NamedCType(goal.name, BaseCType(optionalSymIntArrayRefT)))
+                return f"{argname}.has_value() ? c10::make_optional(c10::asIntArrayRefSlow(*{argname})) : c10::nullopt"
```
Comment (albanD) [inline torchgen/api/translate.py:356]: Why do we try to generate a SymIntArrayRef first?

File: aten/src/ATen/native/Convolution.cpp [hunk 3]
```diff
@@ -8,6 +8,9 @@
 #include <ATen/native/xnnpack/Engine.h>
 #include <ATen/NativeFunctions.h>
 #include <c10/util/accumulate.h>
+#include <ATen/core/ATen_fwd.h>
```
Comment (albanD) [inline aten/src/ATen/native/Convolution.cpp:11]: Why these new includes?

File: tools/autograd/load_derivatives.py (hunk 1 above)
Comment (albanD) [reply tools/autograd/load_derivatives.py:788]: Ho that is subtle, my bad!

**Review by albanD — APPROVED** [review]
Sounds good then.
The translate try/except should do the same thing as before in the worst case and we don't care about codegen speed

**Comment by ezyang** [comment]
Need to also update the ivalue_to_arg logic

**Pull request closed without merging**